sdk/core/azure-core-tracing-opentelemetry/inc/azure/core-tracing-opentelemetry/opentelemetry.hpp

```diff
@@ -0,0 +1,161 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
```

Comment:
Is this header needed at all? I don't see any package using it. And everything it has is in internal namespace.
Maybe, we need a package without any public headers?

Reply:
No package will ever consume this (unless the package is testing its open telemetry integration).

This file defines a single public class: OpenTelemetryProvider, and end-user developers will absolutely use it. All the _internal types are in the header because C++ gets really annoyed when you try to reference a class that is never defined.